neo/rawio/biocamrawio.py

```diff
@@ -122,10 +122,27 @@ def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop, strea
             i_start = 0
         if i_stop is None:
             i_stop = self._num_frames
-        if channel_indexes is None:
-            channel_indexes = slice(None)
+
+
         data = self._read_function(self._filehandle, i_start, i_stop, self._num_channels)
-        return data[:, channel_indexes]
+
+        # older style data return everything
+        if len(data.shape) > 1:
+            if channel_indexes is None:
+                channel_indexes = slice(None)
+            sig_chunk = data[:, channel_indexes]
+        # newer style data returns an initial flat array
+        else:
+            if channel_indexes is None:
+                channel_indexes = [ch for ch in range(self._num_channels)]
+
+            sig_chunk = np.zeros((i_stop-i_start, len(channel_indexes)))
```

Comment:
Can channel_indexes be None or slice here, if so, len will fail.

Reply:
It can't be None (because if None it is converted to a list). It shouldn't be a slice based on what Neo allows is. So I think this should be fine.

I wasn't aiming to refactor the IO for this fix, but I think unifying the reading functions despite their different structures would probably be good longer term. But really only an ArrayLike or None should go into channel_indexes and I force it if it is None.

Reply:
Good. Just pointing out something that I stumbled upon a couple of times.